providers/dell/idrac.go

```diff
@@ -204,21 +205,37 @@ func (c *Conn) Compatible(ctx context.Context) bool {
 
 // PowerStateGet gets the power state of a BMC machine
 func (c *Conn) PowerStateGet(ctx context.Context) (state string, err error) {
+	if err := c.deviceSupported(ctx); err != nil {
```

Comment:
This provider already has this check in `Open`. I'm not following the value add for putting this in all these individual methods. As far as I can tell `deviceSupported` doesnt do any method specific checks either.

Reply:
Thanks for pointing that out, you're right the dell, openbmc checks were redundant. 

I noticed dell showing up as a `successfulProvider` on another device - which was odd, but I haven't been able to reproduce the case, and theres no way it should end up in there. Changes reverted.